prboom2/src/gl_main.c

```diff
@@ -2328,20 +2328,29 @@ static void gld_DrawSprite(GLSprite *sprite)
 
 static void gld_AddHealthBar(mobj_t* thing, GLSprite *sprite)
 {
-  if (((thing->flags & (MF_COUNTKILL | MF_CORPSE)) == MF_COUNTKILL) && (thing->health > 0))
+  if ((thing->flags & MF_SHOOTABLE) && thing->info->spawnhealth > 0 &&
+      thing->health > 0)
   {
     GLHealthBar hbar;
     int health_percent = thing->health * 100 / thing->info->spawnhealth;
 
-    hbar.cm = -1;
-    if (health_percent <= health_bar_red)
-      hbar.cm = CR_RED;
-    else if (health_percent <= health_bar_yellow)
-      hbar.cm = CR_YELLOW;
-    else if (health_percent <= health_bar_green)
-      hbar.cm = CR_GREEN;
+    hbar.r = hbar.g = -1.0f;
```

Comment:
Hmm, I don't think initializing an illegal value is appropriate here, especially when you only check for one illegal value later. While it works today, it could easily break in the future.

I still recommend having a catch-all else case that will initialize the values to something legal but innocuous, and `r=0.0f;g=1.0f;` seems to fit to me.

Others may have differing opinions.

Reply:
Fine, I'll test for both.